wiring/src/spark_wiring_ble.cpp

```diff
@@ -229,23 +238,81 @@ uint8_t BleAddress::operator[](uint8_t i) const {
     return address_.addr[i];
 }
 
-BleAddress& BleAddress::operator=(const hal_ble_addr_t& addr) {
-    address_ = addr;
+BleAddress& BleAddress::operator=(const BleAddress& addr) {
+    address_ = addr.address_;
     return *this;
 }
 
+BleAddress& BleAddress::operator=(const hal_ble_addr_t& addr) {
+    return *this = BleAddress(addr);
+}
+
 BleAddress& BleAddress::operator=(const uint8_t addr[BLE_SIG_ADDR_LEN]) {
+    // We just update the address value while keep the address type
     memcpy(address_.addr, addr, BLE_SIG_ADDR_LEN);
     return *this;
 }
 
+BleAddress& BleAddress::operator=(const char* address) {
+    // We just update the address value while keep the address type
```

Comment:
Any chance this might result in some inconsistencies? Should we parse the address type somehow?

Reply:
We cannot determine the address type for the given string. For example, if the most significant two bits are `0b11`, it can be either a public address or a static address. Same to thee cases that the most significant two bits are `0x00` or `0x01`. I think it's user's responsibility to check the validity using `isValid()` that will be introduced in another PR.